### DISCUSSION TITLE
POESP32

### DISCUSSION THREAD

**Discussion by swalsh11** [raomin/ESPAltherma#366] [General]
I have the m5StickC monitoring my Altherma Monoblock, Wifi signal is poor, even with the ESP32 mounted external to the unit, I think it might be possible to use a U138 POESP32  as an Ethernet adapter to the M5stick, And I could just pull an Ethernet  cable to the heat pump.  this is the unit: https://docs.m5stack.com/en/unit/poesp32, has anyone done this already?  or offer Any tips or advice on how I could proceed.

**Comment by bashers222** [comment]
I have attempted this but abandoned it due to lack of time
I too suffer from poor wifi in the closet where my ESP32 lives so will reattempt it this week
There are a few steps required:

1. In the main.cpp, include the ethernet module "#include "ETH.h" for the card and comment out the wifi
2. Add the ethernet modules in the defs
3. Add the config into platform.ini and main.cpp to activate the PINS ofr networking
4. Remove the section (line 104 to 151) in the main.cpp that checks for wifi connectivty and reboots if its not online. This is obviously unneccessary

**Comment by bgcngm** [comment]
You can take a look at https://github.com/raomin/ESPAltherma/pull/399. It includes support for WT32-ETH01, so you be able to easily adapt to POESP32.

**Reply by swalsh11** [reply]
Thanks for the link I will have a go at getting it to work

**Reply by swalsh11** [reply]
That Complied and Flashed onto the Board with no Errors,  I only edited to use G16 & G17 on the Groove port as RX & TX. as I am not doing any additional Control.  When I plug in the Ethernet cable, it is not getting any lights, I am getting this on the Serial Monitor: 

Attempting MQTT connection...[ 16073][E][WiFiClient.cpp:242] connect(): connect on fd 49, errno: 118, "Host is unreachable"
failed, rc=-2, try again in 5 seconds,

**Reply by swalsh11** [reply]
More Debug output. It is struggling with the ESP_ETH_DRIVER _INSTALL, and failing to initilise the mac address,   

rst:0xc (SW_CPU_RESET),boot:0x12 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:1344
load:0x40078000,len:13924
ho 0 tail 12 room 4
load:0x40080400,len:3600
entry 0x400805f0
Restoring previous state: OffSetting up ethernet...E (1012) esp.emac: emac_esp32_init(349): reset timeout
**E (1012) esp_eth: esp_eth_driver_install(214): init mac failed
[  1015][E][ETH.cpp:321] begin(): esp_eth_driver_install failed**
Connecting to MQTT server...Attempting MQTT connection...[  1031][E][WiFiClient.cpp:242] connect(): connect on fd 49, errno: 118, "Host is unreachable"

**Comment by Alibloke** [comment]
@swalsh11 More work needs to be done for the POESP32 as it uses the IP101 PHY.  I've managed to get it working, will submit a pull request shortly.

**Reply by swalsh11** [reply]
Awesome, Thank you.

**Reply by swalsh11** [reply]
Thanks For the Lead, I figured it out from this: https://github.com/espressif/arduino-esp32/issues/3554
I added in 3 new line 18,19,20 in main.cpp: 
```
 #include <ETH.h>
 #define ETH_ADDR        1
 #define ETH_POWER_PIN   5
 #define ETH_TYPE        ETH_PHY_IP101
```

I  edited line 215 in Main.cpp to: 
`ETH.begin(ETH_ADDR, ETH_POWER_PIN, ETH_PHY_MDC, ETH_PHY_MDIO, ETH_TYPE);`
and I edited setup.h for:
```
 #if defined(WT32_ETH01);
#define RX_PIN    16// Pin connected to the TX pin of X10A
#define TX_PIN    17// Pin connected to the RX pin of X10A
```

**Comment by Alibloke** [comment]
That's pull request #400 submitted. @swalsh11 yes that's essentially what I did, though I added MDC & MDIO, not sure if they're needed or not, but it worked.

**Reply by swalsh11** [reply]
Thanks Alibloke

**Comment by Alibloke** [comment]
OK that's pull request #401 submitted and #400 deleted.  @bgcngm hopefully I did it right this time.
